src/api/users/services/addBookUpvote.ts

```diff
@@ -0,0 +1,19 @@
+import {  UserModel, BookUpvoteModel } from "../../../models";
+import { UserNotFound, BookUpvoteNotFound } from "../../../errors";
+
+export default async function addBookUpvote(userId: string, upvoteId: string): Promise<void> {
+    try {
+      const user = await UserModel.findById(userId);
+      if (!user) throw new UserNotFound();
+    
+      const upvote = await BookUpvoteModel.findById(upvoteId);
+      if (!upvote) throw new BookUpvoteNotFound();
+
```

Comment:
Let's check whether bookUpvoteId is in user.bookUpvoteIds and return new RealtionshipAlreadyExists() if so